src/Controllers/BatchProcessorController.ts

```diff
@@ -45,36 +46,49 @@ module StreamStats.Controllers {
         public selectedBatchProcessorTabName: string;
         public displayMessage: string;
         public isValid: boolean;
-        public AppVersion: string;  
+        public AppVersion: string;
         public submitBatchInfo: string;
         public regionList: Object;
         public flowStatsList: Array<any>;
         public selectedRegion: string;
         public flowStatsAllChecked: boolean;
-        public regionStatsList: Array<any>;
+        public selectedFlowStatsList: Array<any>;
+        public availableParamList: Array<any>;
+        public flowStatChecked: boolean;
+        public selectedParamList: Array<any>
+        public parametersAllChecked: boolean;
+        public flowStatisticsAllChecked: boolean;
+        public regionParamList: Array<any>;
+        public showBasinCharacterstics: boolean;
 
         //Constructor
         //-+-+-+-+-+-+-+-+-+-+-+-
         static $inject = ['$scope', '$http', 'StreamStats.Services.ModalService', 'StreamStats.Services.nssService', '$modalInstance'];
-        constructor($scope: IBatchProcessorControllerScope, $http: ng.IHttpService, modalService: Services.IModalService, nssService: Services.InssService, modal:ng.ui.bootstrap.IModalServiceInstance) {
+        constructor($scope: IBatchProcessorControllerScope, $http: ng.IHttpService, modalService: Services.IModalService, nssService: Services.InssService, modal: ng.ui.bootstrap.IModalServiceInstance) {
             super($http, configuration.baseurls.StreamStats);
             $scope.vm = this;
             this.modalInstance = modal;
             this.modalService = modalService;
             this.selectedBatchProcessorTabName = "submitBatch";
             this.nssService = nssService;
-            this.regionStatsList = [];
-            this.flowStatsAllChecked = true;
-            this.init();  
-        }  
-        
+            this.selectedFlowStatsList = [];
+            this.selectedParamList = [];
+            this.availableParamList = [];
+            this.regionParamList = [];
+            // this.flowStatsAllChecked = true;
+            this.flowStatChecked = false;
+            this.parametersAllChecked = true;
+            this.flowStatisticsAllChecked = true;

```

Comment:
I recommend initializing these to `false`, because when we open the modal, the parameters and flowStatistics are _not_ all checked. (If you make this change, be sure to change the associated logic in the controller.)
```suggestion
            this.parametersAllChecked = false;
            this.flowStatisticsAllChecked = false;

```